sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -180,4 +184,60 @@ export class ApplicationController extends BaseController {
       );
     }
   }
+
+  /**
+   * Generic method to update all Student Application status from frontend.
+   * @param applicationId application id to be updated.
+   * @body payload contains the status, that need to be updated
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Patch(":applicationId/update-status")
```

Comment:
IMO,  @Patch("/status/:applicationId") is preferred REST endpoint

Reply:
I guess, we have similar controller endpoints, we have a preference of passing the applicationId before the type of process the controller does.

Some Examples:
============
@Patch(":applicationId/submit")
@Get(":applicationId/assessment")
@Patch(":applicationId/confirm-assessment")

Reply:
IMO, :applicationId should come after `application/`. @andrewsignori-aot  @guru-aot  thoughts?

Reply:
@dheepak-aot the main controller has the /application so all the other endpoints will be

@patch(":applicationId/submit") - /application/:applicationId/submit
@get(":applicationId/assessment") - /application/:applicationId/assessment
@patch(":applicationId/confirm-assessment") - /application/:applicationId/confirm-assessment

I guess, you might have missed the appended controller endpoint.

Now I feel, it make sense after appending the /application. Am I right? Also its the preferred way of REST definition.

Reply:
@ann-aot @guru-aot Agreed that :applicationId  should come after application. My understand was wrong about this controller initially.
But still verb(s) shouldn't be part of URI. like update, or create. IMO @Patch(":applicationId/status") will be better.

Reply:
I agree, patch(":applicationId/status"). I think you are right @dheepak-aot

Reply:
updated @dheepak-aot

Reply:
Thanks @ann-aot. Great work overall